packages/concerto-core/lib/introspect/modelfile.js

```diff
@@ -275,19 +275,20 @@ class ModelFile extends Decorated {
         // Validate all of the types in this model file.
         // Check if names of the declarations are unique.
         const uniqueNames = new Set();
-        this.declarations.forEach(
-            d => {
-                const fqn = d.getFullyQualifiedName();
-                if (!uniqueNames.has(fqn)) {
-                    uniqueNames.add(fqn);
-                } else {
-                    throw new IllegalModelException(
-                        `Duplicate class name ${fqn}`
-                    );
-                }
+        const concertoTypes = new Set(['Concept', 'Asset', 'Transaction', 'Participant', 'Event']);
+        this.declarations.forEach(d => {
+            const fqn = d.getFullyQualifiedName();
+            const shortName = ModelUtil.getShortName(fqn);
+            if (uniqueNames.has(fqn)) {
+                throw new IllegalModelException(`Duplicate class name: ${fqn}`);
             }
-        );
-
+            if (!concertoTypes.has(shortName) && this.importShortNames.has(shortName)) {
+                throw new IllegalModelException(
+                    `TypeName Conflict: The type '${shortName}' is being redefined, but it already exists as an imported type from '${this.importShortNames.get(shortName)}'. Please rename your local type or avoid importing conflicting types.`
+                );
+            }
```

Comment:
What would be the short name if this was an aliased type? is that accounted for?

Reply:
Yes it is accounted for . 
imporShortNames is map from local type to fqn . So if it was a aliased name then it is registered in the importShortNames map